framework/di/Container.php

```diff
@@ -527,7 +531,17 @@ public function resolveCallableDependencies(callable $callback, $params = [])
                 } elseif (isset(Yii::$app) && Yii::$app->has($name) && ($obj = Yii::$app->get($name)) instanceof $className) {
                     $args[] = $obj;
                 } else {
-                    $args[] = $this->get($className);
+                    // If the argument is optional we catch not instantiable exceptions.
+                    try {
+                        $args[] = $this->get($className);
+                    } catch (NotInstantiableException $e) {
```

Comment:
It's not just interfaces. It is anything that is not instantiable, so that includes abstract classes as well.
There is no contradiction here, there are just different situations:
1. Required dependency. --> Exception on misconfiguration.
2. Optional dependency. --> Silently pass default value (which is always `NULL` when we talk about classes / interfaces).

When there is an optional dependency, we want to fill it if we can, or skip it if we cannot fill it.